yarn-project/circuit-types/src/l2_block.ts

```diff
@@ -688,6 +565,18 @@ export class L2Block {
     }
   }
 
+  private assertLogsAttached() {
+    if (!this.areLogsAttached()) {
+      throw new Error(
+        `newEncryptedLogs and newUnencryptedLogs must be defined (block ${this.header.globalVariables.blockNumber})`,
+      );
+    }
+  }
+
+  private areLogsAttached() {
```

Comment:
Again, this could just be a function on the body. A lot of the checks here are only on the body, so we can move it into the body instead of having it in the block 👀